src/Compilers/Core/Portable/Text/CompositeText.cs

```diff
@@ -373,5 +378,131 @@ private static void TrimInaccessibleText(ArrayBuilder<SourceText> segments)
                 segments.Add(writer.ToSourceText());
             }
         }
+
+        /// <summary>
+        /// Delegates to SourceTexts within the CompositeText to determine line information.
+        /// </summary>
+        private sealed class CompositeLineInfo : TextLineCollection
+        {
+            private readonly CompositeText _compositeText;
+
+            // The starting line number for the correspondingly indexed SourceTexts in _compositeText.Segments
+            private readonly int[] _segmentLineNumbers;
```

Comment:
if this is constructed and initialized in constructor, make into an IMmutableArray.  no need to allow code to accidentally mtuate.

Reply:
makes sense, was just trying to match what CompositeText._segmentOffsets has.